### ISSUE TITLE
[Discuss] Should we allow users to disable collection of metrics

### ISSUE THREAD

**Issue by ph** [elastic/logstash#6298]
@L4rS6 discovered an issue with our metric collection; it doesn't work when procfs is not mounted.
For security reason users can disable the access of procfs, this restriction will make logstash crash at startup. 

For me, its a bug and we should resolve #6296, but let's open the discussion should we allow users to be able to disable the metric collection?  Because right now 5.0 is not usable in that scenario and there is no workaround until we fixed the issue.

**Comment by jsvd** [comment]
@ph you mean https://github.com/elastic/logstash/commit/7132c68cf4dd698c2f55243ef0504dc58bfa5bd9 ?

**Comment by untergeek** [comment]
I thought they could be disabled. I was just thinking of a as yet unreleased feature.

**Comment by ph** [comment]
@jsvd Yes, but officially expose it in the `logstash.yml`, the pipeline now support it. But we only use that feature for the monitoring pack.

**Comment by jsvd** [comment]
yeah, I'm +1 on documenting this setting's existence + making sure the web api doesn't freak out when metrics aren't collected.

**Comment by suyograo** [comment]
I personally don't like disabling monitoring information for the entire product. That said, we shouldn't crash when we can't gather any metrics -- we could just log a warning and continue. Other metrics which can be gathered could be very useful. Exposing the ability to disable metrics will mean that at the slightest performance issue, users would first disable this, and we are back to a black box. I think LS should be resilient to failures in collecting metrics and should start normally.

**Comment by markwalkom** [comment]
https://discuss.elastic.co/t/how-disable-logstash-monitoring-apis/81635 is relevent
